indico/modules/users/controllers.py

```diff
@@ -435,6 +445,28 @@ def _process(self):
         return jsonify(self.user.settings.get('use_markdown_for_minutes'))
 
 
+class RHUserPreferencesMastodonServer(RHUserBase):
+    @use_kwargs({
+        'server_url': fields.String(required=True),
+    })
+    def _process_POST(self, server_url):
+        if not social_settings.get('enabled'):
+            raise Forbidden('The social share widget is not enabled.')
+
+        url = urlsplit(server_url)
+        if url.scheme not in ('http', 'https'):
+            raise ExpectedError(_('Invalid URL.'))
+        url = strip_path_from_url(urlunsplit(url))
```

Comment:
Similar to my other comment, there's no need for this roundtrip

```suggestion
        url = strip_path_from_url(server_url)
```